geoutils/georaster/raster.py

```diff
@@ -2487,3 +2487,110 @@ def polygonize(
         gdf.set_crs(self.crs, inplace=True)
 
         return gv.Vector(gdf)
+
+    def proximity(
+        self,
+        vector: Vector | None = None,
+        target_values: list[float] | None = None,
+        geometry_type: str = "boundary",
+        in_or_out: Literal["in"] | Literal["out"] | Literal["both"] = "both",
+        distance_unit: Literal["pixel"] | Literal["georeferenced"] = "georeferenced",
+    ) -> Raster:
+        """
+        Proximity to this Raster's target pixels, or to a Vector's geometry, computed for each cell of this Raster's grid.
+
+        When passing a Vector, by default, the boundary of the geometry will be used. The full geometry can be used by
+        passing "geometry", or any lower dimensional geometry attribute such as "centroid", "envelope" or "convex_hull".
+        See all geometry attributes in the Shapely documentation at https://shapely.readthedocs.io/.
+
+        :param vector: Vector for which to compute the proximity to geometry,
+            if not provided computed on this Raster target pixels.
+        :param target_values: (Only with Raster) List of target values to use for the proximity,
+            defaults to all non-zero values.
+        :param geometry_type: (Only with a Vector) Type of geometry to use for the proximity, defaults to 'boundary'.
+        :param in_or_out: (Only with a Vector) Compute proximity only 'in' or 'out'-side the geometry, or 'both'.
+        :param distance_unit: Distance unit, either 'georeferenced' or 'pixel'.
+
+        :return: Proximity raster.
+        """
+
+        proximity = proximity_from_vector_or_raster(raster=self, vector=vector, target_values=target_values, geometry_type=geometry_type,
+                                                    in_or_out=in_or_out, distance_unit=distance_unit)
+
+        return self.copy(new_array=proximity)
+
+
+# -----------------------------------------
+# Additional stand-alone utility functions
```

Comment:
At some point, we re-organized raster.py because the idea was to have the Raster class in a separate file than the other functions, see [the issue](https://github.com/GlacioHack/geoutils/issues/215).
Unfortunately, we never went all the way, because there is no other file in the georaster folder. spatial_tools.py most likely belongs there, but maybe it should be renamed and the functions double-checked. This is related to [this opened issue](https://github.com/GlacioHack/geoutils/issues/311).

Reply:
Ah yes, that was why... Good point.
This will have to be done separately during re-structuration, especially when we address `spatial_tools.py` so I leave it out of this PR for the moment.